### PR TITLE
changes to pipe

### PR DESCRIPTION
**Pull request by MateiHMCTS**
Changes to pipe as it seems that it was returning null in some cases and this seemed to also triggering change detection which seemed to be looping in some cases

### CODE REVIEW

File: projects/rpx-xui-translation/src/lib/rpx-translation.service.ts [hunk 1]
```diff
@@ -109,8 +108,9 @@ export class RpxTranslationService {
             this.phrases[phrase].next(`${phrase} [Translation in progress]`);
             this.load(phrase, lang, yesOrNo);
           }
-        })
-      ).subscribe(() => { });
+        }),
+        pipe(take(1))
```
Comment (Daniel-Lam) [inline projects/rpx-xui-translation/src/lib/rpx-translation.service.ts:112]: @MateiHMCTS - any reason for the double `pipe`? Isn't there already one from Line 99?

File: projects/rpx-xui-translation/src/lib/rpx-translate.pipe.ts [hunk 2]
```diff
@@ -28,10 +29,14 @@ export class RpxTranslatePipe extends AsyncPipe implements PipeTransform {
...
       }
-      const ret = super.transform<string>(o);
+
+      const ret = this.asyncPipe.transform<string>(o);
       return ret as unknown as T;
     }
     return null;
   }
 
+  ngOnDestroy() {
```
Comment (Daniel-Lam) [inline projects/rpx-xui-translation/src/lib/rpx-translate.pipe.ts:39]: @MateiHMCTS - missing access modifier and return type

**Review by ronaldmansveld — CHANGES_REQUESTED** [review]
Why these changes to the pipe? What are you trying to achieve here?

File: projects/rpx-xui-translation/src/lib/rpx-translate.pipe.ts [hunk 3]
```diff
@@ -8,13 +8,14 @@ import { Replacements, RpxTranslationService } from './rpx-translation.service';
...
+export class RpxTranslatePipe implements PipeTransform, OnDestroy {
+  private asyncPipe: AsyncPipe;
 
   constructor(
     private translationService: RpxTranslationService,
-    ref: ChangeDetectorRef
+    private injector: Injector
   ) {
-    super(ref);
+    this.asyncPipe = new AsyncPipe(injector.get(ChangeDetectorRef));
```
Comment (Daniel-Lam) [inline projects/rpx-xui-translation/src/lib/rpx-translate.pipe.ts:18]: @MateiHMCTS - query about why this pipe no longer extends the `AsyncPipe` but I think @ronaldmansveld is going to comment on this

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@MateiHMCTS - raised some observations; also, I think @ronaldmansveld will be providing some input into this, too

File: projects/rpx-xui-translation/src/lib/rpx-translation.service.ts (hunk 1 above)
Comment (MateiHMCTS) [reply projects/rpx-xui-translation/src/lib/rpx-translation.service.ts]: Forgot to remove it

**Comment by MateiHMCTS** [comment]
> Why these changes to the pipe? What are you trying to achieve here?

Please look at this PR as an example for the issue that made me look into this
https://github.com/hmcts/rpx-xui-webapp/pull/3019
It is just a fork of the original PR https://github.com/hmcts/rpx-xui-webapp/pull/3012 -- only change is the rpx-xui-translation lib version change (the original PR has 0.2.3 which is the solution proposed; the forked PR has the latest package -- 0.2.2-yes-no-translations-fix)


If you go to Create Case - Jurisidction: Manage probate application; Case type: grant of representation; Event: Apply for probate;
Then Continue until you get to the form
once you get to the form press Save and continue to get the validation errors

You should see two issues:
	Page-level validation errors flickering
	Field-level validation errors being null (i.e. not shown)  You can also check the logs and see it changes between null and the “whatever” value there should be



I think what happens is that when we call super.transform; AsyncPipe’s transform calls the parent which calls this.transform if the obj passed is not the same (i.e. doesn’t have the same reference) the AsyncPipe.transform method calls this.trasnform. At that point it should call super.transform, but it calls this.transform; this is the RpxtranslationPipe class, so RpxTranslationPipe’s transform method is called. The latter checks if it’s a string, but the AsyncPipe this.transform passes an observable as an argument, so RpxTranslationPipe.transform returns null as it fals the if statement (i.e. value is not a string). Then it’s only when the next change detection is triggered, it tries again;
I am not super happy about the solution I have either as I feel like there still might be some race condition there, but it seems to work for now?


Also, ngOnDestroy on parent is never called as we don’t do super.ngOnDestroy in the extended class — it doesn’t do this automatically, so this creates a memory leak. This isn’t a big deal though as we can easily just add super.ngOnDestroy on ngOnDestroy.

**Review by Daniel-Lam — APPROVED** [review]

**Comment by MateiHMCTS** [comment]
Closing PR as these changes are already included in https://github.com/hmcts/rpx-xui-translation/pull/21

**Pull request closed without merging**